flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java

```diff
@@ -423,14 +430,15 @@ private void parseAndSetRequiredProperties() {
                 startingOffsetsInitializer.getAutoOffsetResetStrategy().name().toLowerCase(),
                 true);
 
-        // If the source is bounded, do not run periodic partition discovery.
-        if (maybeOverride(
-                KafkaSourceOptions.PARTITION_DISCOVERY_INTERVAL_MS.key(),
-                "-1",
-                boundedness == Boundedness.BOUNDED)) {
-            LOG.warn(
-                    "{} property is overridden to -1 because the source is bounded.",
-                    KafkaSourceOptions.PARTITION_DISCOVERY_INTERVAL_MS);
+        // If the source is bounded or stoppingOffsetsInitializer is specified, do not run periodic
```

Comment:
Should we update the comments here as well? It probably should be something like:
"We only disable the periodic partition discovery in the following cases:
1. The source is bounded.
2. The source has stopping offsets and there is no explicit partition discovery interval setting."

Reply:
I think these should be exposed to users in the official documents